src/main/java/org/folio/rs/service/RetrievalQueueService.java

```diff
@@ -96,4 +138,52 @@ private void saveRetrievalQueueWithCurrentDate(RetrievalQueueRecord record) {
     return (record, criteria, builder) -> builder.equal(record.get(ID), stringToUUIDSafe(id));
   }
 
+  private LocationMapping getLocationMapping(Item item) {
+    return Objects.nonNull(item.getEffectiveLocation())
+        ? locationMappingsService.getMappingByFolioLocationId(item.getEffectiveLocation().getId())
+        : null;
+  }
+
+  private Item getOriginalItemByBarcode(MovedEventRequest movedEventRequest) {
+    ResultList<Item> items = inventoryClient.getItem("barcode==" + movedEventRequest.getItemBarCode());
+    if (Objects.isNull(items)) {
+      throw new EntityNotFoundException("Item with barcode " + movedEventRequest.getItemBarCode() + NOT_FOUND);
+    }
+    return items.getResult().get(0);
+  }
+
+  private User getUserByRequesterId(MovedEventRequest movedEventRequest) {
+    ResultList<User> users = usersClient.query("id==" + movedEventRequest.getRequesterId());
+    if (Objects.isNull(users)) {
```

Comment:
Double check please this functionality in case if users
will be empty but dont null